Dialog/SpritesEditorDialog.cpp

```diff
@@ -0,0 +1,561 @@
+{
+    ui->setupUi(this);
+
+    // Render Setup
+    SpriteTileMAPScene = new QGraphicsScene(0, 0, 8 * 32, 8 * 32);
+    ui->graphicsView_SpriteTileMap->setScene(SpriteTileMAPScene);
+    ui->graphicsView_SpriteTileMap->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+    ui->graphicsView_SpriteTileMap->scale(2, 2);
+    ui->graphicsView_SpriteTileMap->SetCurrentSpritesEditor(this);
+    SpritesetTileMAPScene = new QGraphicsScene(0, 0, 8 * 32, 8 * 32);
+    ui->graphicsView_SpritesetTileMap->setScene(SpritesetTileMAPScene);
+    ui->graphicsView_SpritesetTileMap->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+    PaletteBarScene = new QGraphicsScene(0, 0, 16 * 8, 16 * 8);
+    // TODO: PaletteBarScene->setBackgroundBrush();
+    ui->graphicsView_SpritePals->setScene(PaletteBarScene);
+    ui->graphicsView_SpritePals->setAlignment(Qt::AlignTop | Qt::AlignLeft);
+    ui->graphicsView_SpritePals->scale(2, 2);
+    ui->graphicsView_SpritePals->SetCurrentSpritesEditor(this);
+
+    // Seems the spinboxes won't trigger the valuechanged event when loading UI
+    // We need to load graphics manually
+    RenderSpritesTileMap();
+    SetSelectedSpriteTile(0);
+    RenderSpritesPalette();
+    SetSelectedEntityColorId(0);
+    RenderSpritesetTileMapAndResetLoadTable();
+
+    // Add ToolTip Text
+    ui->lineEdit_SpritesetLoadTable->setToolTip(tr("Non-universal Entities can only be loaded start from tile line 16 and palette line 8.\n"
+                                                   "Every entity use 2 bytes, the first byte is global entity id,\n"
+                                                   "the second one is the offset used to calculate the address to DMA the palettes and tiles data into the ram.\n"
+                                                   "According to the vanilla game engine, the previous loaded palettes and tiles can be overwritten by the others'\n"
+                                                   "So set the load table carefully.\n"
+                                                   "The editor use different palettes for every entity, but ingame, some of them share the same palettes and tiles\n"
+                                                   "you will find the editor act differently from the real game play.\n"
+                                                   "In this case, you need to modify all of the entities who sharing the same tiles and palettes."));
+}
+
+/// <summary>
+/// Deconstructor of SpritesEditorDialog class.
+/// </summary>
+SpritesEditorDialog::~SpritesEditorDialog()
+{
+    delete ui;
+}
+
+void SpritesEditorDialog::SetSelectedEntityColorId(int colorID)
+{
+    if (colorID > 15) colorID = 15;
+    curEntityColorIdInPalette = colorID;
+    SelectionBox_Color->setPos(colorID * 8, curEntityPalId * 8);
+    SelectionBox_Color->setVisible(true);
+    QColor color = GetCurEntityPtr()->GetPalette(curEntityPalId)[colorID];
+    ui->label_CurColorValue->setText(QString("RGB888: (") +
```

Comment:
Let's include the option to view this as hex. Perhaps the solution involves adding a toggle in the UI.

Reply:
done.